awspub/configmodels.py

```diff
@@ -94,6 +95,34 @@ class ConfigImageSSMParameterModel(BaseModel):
     )
 
 
+class ConfigSNSNotificationProtocol(str, Enum):
+    DEFAULT = "default"
+    EMAIL = "email"
+
+
+class ConfigImageSNSNotificationModel(BaseModel):
+    """
+    Image/AMI SNS Notification specific configuration to notify subscribers about new images availability
+    """
+
+    model_config = ConfigDict(extra="forbid", use_enum_values=True)
+
+    subject: str = Field(description="The subject of SNS Notification", min_length=1, max_length=99)
+    message: Dict[ConfigSNSNotificationProtocol, str] = Field(
```

Comment:
i'd prefer to separate this into two fields. e.g. `notification_protocol` and `message`. i don't see a use case where you wouldn't supply either (assuming this model is supplied in the config)

Reply:
the message can be different by protocol. so eg:
```
{
  "default": "the default message",
  "email": "the email message",
  "email-json": "the email json message",
}
```

how would that look like with separated fields?

Reply:
i've looked through the [publish API](https://boto3.amazonaws.com/v1/documentation/api/1.35.6/reference/services/sns/client/publish.html) and agree. this seems to be fine as is.